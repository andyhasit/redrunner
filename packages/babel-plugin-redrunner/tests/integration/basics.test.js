const {c, EOL, transform} = require('../utils')


test('Empty div created correct HTML', () => {
  const src = `
  class MyComponent extends Component {
    __html__ = '<div/>'
  }`
  expect(transform(src)).toMatchSnapshot()
});


test('__clone__ instead of __html__', () => {
  const src = `
  class MyComponent extends Component {
    __clone__ = '<div/>'
  }`
  expect(transform(src)).toMatchSnapshot()
});
