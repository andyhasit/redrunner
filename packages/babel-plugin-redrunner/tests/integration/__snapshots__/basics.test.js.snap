// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__clone__ instead of __html__ 1`] = `
"class MyView extends View {}

var _MyViewPrototype = MyView.prototype;
var __bu__ = _MyViewPrototype.__bu;
_MyViewPrototype.__ht = '<div></div>';
_MyViewPrototype.__wc = [];
_MyViewPrototype.__qc = __bu__._qc({});
_MyViewPrototype.__ip = {};

_MyViewPrototype.__bv = function (view, prototype) {
  view.__bd(prototype, true);

  view.dom = {};
};

_MyViewPrototype.__cn = undefined;"
`;

exports[`Empty div created correct HTML 1`] = `
"class MyView extends View {}

var _MyViewPrototype = MyView.prototype;
var __bu__ = _MyViewPrototype.__bu;
_MyViewPrototype.__ht = '<div></div>';
_MyViewPrototype.__wc = [];
_MyViewPrototype.__qc = __bu__._qc({});
_MyViewPrototype.__ip = {};

_MyViewPrototype.__bv = function (view, prototype) {
  view.__bd(prototype, false);

  view.dom = {};
};"
`;
